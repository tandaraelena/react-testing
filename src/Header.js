import React from "react";

export default function Header({ content }) {
  return (
    <header>
      <h3 data-testid="h3id" className="fancy-h3">
        {content}
      </h3>
    </header>
  );
}
