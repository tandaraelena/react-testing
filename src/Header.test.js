import React from 'react'
import {render, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Header from './Header'

// after each test afterEach(cleanup) will clean things up
afterEach(cleanup);

// will create a snapshot of Header component and if we refactor
// the code and change the elements of this comp, this test will fail
it('renders', () => {
  const { asFragment } = render(<Header content="Hello" />);
  expect(asFragment()).toMatchSnapshot();
});

it('inserts content in h3', () => {
  const { getByTestId, getByText } = render(<Header content="Hello" />);
  
  expect(getByTestId('h3id')).toHaveTextContent("Hello");
  expect(getByText('Hello')).toHaveClass("fancy-h3");
})