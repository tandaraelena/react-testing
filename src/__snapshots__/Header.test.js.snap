// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <header>
    <h3
      class="fancy-h3"
      data-testid="h3id"
    >
      Hello
    </h3>
  </header>
</DocumentFragment>
`;
